lib/gnat.ex

```diff
@@ -25,20 +25,33 @@ defmodule Gnat do
   def pub(pid, topic, message, opts \\ []), do: GenServer.call(pid, {:pub, topic, message, opts})
 
   @doc """
-  Send a request listen for response(s)
+  Send a request and listen for a response synchronously
 
   Following the nats [request-response pattern](http://nats.io/documentation/concepts/nats-req-rep/) this
   function generates a one-time topic to receive replies and then sends a message to the provided topic.
 
   Supported options:
-    * max_messages: how many messages to expect before cleaning up the inbox subscription (default `1`)
-    * recipient: which pid should receive the responses (default `self()`)
+    * receive_timeout: an integer number of milliseconds to wait for a response. Defaults to 60_000
+
+  ```
+  {:ok, gnat} = Gnat.start_link()
+  case Gnat.request("i_can_haz_cheezburger", "plZZZZ?!?!?") do
+    {:ok, %{body: delicious_cheezburger}} -> :yum
+    {:error, :timeout} -> :sad_cat
+  end
+  ```
   """
   def request(pid, topic, body, opts \\ []) do
-    recipient = Keyword.get(opts, :recipient, self())
-    max_messages = Keyword.get(opts, :max_messages, 1)
+    receive_timeout = Keyword.get(opts, :receive_timeout, 60_000)
     inbox = "INBOX-#{:crypto.strong_rand_bytes(12) |> Base.encode64}"
-    GenServer.call(pid, {:request, %{recipient: recipient, max_messages: max_messages, inbox: inbox, body: body, topic: topic}})
+    {:ok, subscription} = GenServer.call(pid, {:request, %{recipient: self(), inbox: inbox, body: body, topic: topic}})
+    receive do
+      {:msg, %{topic: ^inbox}=msg} -> {:ok, msg}
+      other -> IO.inspect(other)
```

Comment:
Derp, I put this in when debugging. I don't expect it to be hit, but if anyone ever sent us a message while we were waiting for a response it would trigger this clause and could mess up some other piece of code that was expecting a message. I removed this in the latest commit